controller/src/beerocks/master/son_management.cpp

```diff
@@ -2071,6 +2071,18 @@ void son_management::handle_bml_message(Socket *sd,
             LOG(ERROR) << "Failed building message cACTION_BML_CHANNEL_SCAN_START_SCAN_RESPONSE !";
             return;
         }
+
+        // Get scan status
+        auto scan_in_progress =
+            database.get_channel_scan_in_progress(request->scan_params().radio_mac, true);
+        if (scan_in_progress) {
+            LOG(ERROR) << "Single scan is still running!";
```

Comment:
Who said its a single scan which is running? Can't it be in continues scan?

Reply:
1. this flow is for single-scan only
2. the single-scan and continuous-scan have separate configuration params in DB - thus a continuous scan in progress should not block a single scan configuration. Moreover - the received single-scan request will now be pending and triggered only after the continuous scan is complete - but this is just elaboration on the flow and not related to the issue described :-)

Reply:
then, 
 either use define "SINGLE_CHANNEL_SCAN" instead of **true** in database.get_channel_scan_in_progres(...)
or/and  add comments 
or/and rename scan_in_progress to  `single_scan_in_progress `

Reply:
edited the comment to `Get single scan status`
and
edited the `scan_in_progress` to `single_scan_in_progress`